File: android/src/main/java/io/parity/signer/components/networkicon/jdenticon/jdenticon_kotlin/ColorTheme.kt
package io.parity.signer.components.networkicon.jdenticon.jdenticon_kotlin

internal fun colorTheme(hue: Float, config: Config) : List<String> {
    return listOf(
        // Dark gray
        Color.hsl(0f, 0f, config.grayscaleLightness(0f)),
        // Mid color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(0.5f)),
        // Light gray
        Color.hsl(0f, 0f, config.grayscaleLightness(1f)),
        // Light color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(1f)),
        // Dark color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(0f))
    )
}

File: ios/Packages/Jdenticon/Sources/Jdenticon/Colors/ColorTheme.swift
//
//  ColorTheme.swift
//
//
//  Created by Krzysztof Rodak on 22/08/2023.
//

import Foundation

func colorTheme(hue: CGFloat, config: Config) -> [String] {
    [
        // Dark gray
        Color.hsl(0, 0, config.grayscaleLightness(0)),
        // Mid color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(0.5)),
        // Light gray
        Color.hsl(0, 0, config.grayscaleLightness(1)),
        // Light color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(1)),
        // Dark color
        Color.correctedHsl(hue, config.saturation, config.colorLightness(0))
    ]
}
